.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+name: Release New Version
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  pipeline:
+    name: Pipeline
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create new GitHub tag
+          id: new_tag
+          uses: butlerlogic/action-autotag@stable
+          with:
+            GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+            strategy: regex
+            root: "version"
+            regex_pattern: "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
```

Comment:
Love it 😊 could we add a comment around what this regex is doing?

Reply:
Yeah....I just copied that regex from the action. It is regex for practically any type of semver possible. Since we only need `x.y.z`, I'm going to simplify it a lot more in the next commit.